druid-shell/src/keyboard.rs

```diff
@@ -679,6 +740,159 @@ impl fmt::Debug for KeyModifiers {
     }
 }
 
+impl From<KeyModifiers> for ModState {
+    fn from(src: KeyModifiers) -> ModState {
+        match (src.alt, src.ctrl, src.meta, src.shift) {
+            (false, false, false, false) => ModState::None,
+            (true, false, false, false) => ModState::Alt,
+            (false, true, false, false) => ModState::Ctrl,
+            (false, false, true, false) => ModState::Meta,
+            (false, false, false, true) => ModState::Shift,
+            (true, true, false, false) => ModState::AltCtrl,
+            (true, false, true, false) => ModState::AltMeta,
+            (true, false, false, true) => ModState::AltShift,
+            (false, true, true, false) => ModState::CtrlMeta,
+            (false, true, false, true) => ModState::CtrlShift,
+            (false, false, true, true) => ModState::MetaShift,
+            (true, true, true, false) => ModState::AltCtrlMeta,
+            (true, false, true, true) => ModState::AltMetaShift,
+            (true, true, false, true) => ModState::AltCtrlShift,
+            (false, true, true, true) => ModState::CtrlMetaShift,
+            (true, true, true, true) => ModState::AltCtrlMetaShift,
+        }
+    }
+}
+
+impl From<ModState> for KeyModifiers {
+    fn from(src: ModState) -> KeyModifiers {
+        let (alt, ctrl, meta, shift) = match src {
+            ModState::None => (false, false, false, false),
+            ModState::Alt => (true, false, false, false),
+            ModState::Ctrl => (false, true, false, false),
+            ModState::Meta => (false, false, true, false),
+            ModState::Shift => (false, false, false, true),
+            ModState::AltCtrl => (true, true, false, false),
+            ModState::AltMeta => (true, false, true, false),
+            ModState::AltShift => (true, false, false, true),
+            ModState::CtrlMeta => (false, true, true, false),
+            ModState::CtrlShift => (false, true, false, true),
+            ModState::MetaShift => (false, false, true, true),
+            ModState::AltCtrlMeta => (true, true, true, false),
+            ModState::AltMetaShift => (true, false, true, true),
+            ModState::AltCtrlShift => (true, true, false, true),
+            ModState::CtrlMetaShift => (false, true, true, true),
+            ModState::AltCtrlMetaShift => (true, true, true, true),
+        };
+        KeyModifiers {
+            alt,
+            ctrl,
+            meta,
+            shift,
+        }
+    }
+}
+
+impl std::cmp::PartialEq<(&'static str, ModState)> for KeyEvent {
```

Comment:
Is this really eq? I think I would be slightly more comfortable with it just being a function, like `key.matches("a", ModState::Shift)` or whatnot.